spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/MetricsEndpoint.java

```diff
@@ -112,15 +112,20 @@ private Tag parseTag(String tag) {
 		return Tag.of(parts[0], parts[1]);
 	}
 
-	private void collectMeters(List<Meter> meters, MeterRegistry registry, String name,
-			Iterable<Tag> tags) {
+	private boolean findFirstMatchingMeters(List<Meter> meters, MeterRegistry registry,
+			String name, Iterable<Tag> tags) {
 		if (registry instanceof CompositeMeterRegistry) {
-			((CompositeMeterRegistry) registry).getRegistries()
-					.forEach((member) -> collectMeters(meters, member, name, tags));
+			return ((CompositeMeterRegistry) registry).getRegistries().stream()
+					.anyMatch((r) -> findFirstMatchingMeters(meters, r, name, tags));
 		}
 		else {
-			meters.addAll(registry.find(name).tags(tags).meters());
+			Collection<Meter> meterFound = registry.find(name).tags(tags).meters();
+			if (meterFound != null && meterFound.size() > 0) {
```

Comment:
Micrometer guarantees that `meterFound` is non-null. Could use `isEmpty()` on second condition.